File: leetcode/src/main/java/leetcode/editor/cn/存在重复元素IIcontainsDuplicateIi.java
/**
 * @Title: 219.存在重复元素 II
 * @TitleSlug: containsDuplicateIi
 * @Author: Neo
 * @Date: 2024-04-29 15:45:19
 */
package leetcode.editor.cn;

import java.util.HashSet;
import java.util.Set;

public class 存在重复元素IIcontainsDuplicateIi {
    public static void main(String[] args) {
        // 测试代码
        // Solution solution = new 存在重复元素IIcontainsDuplicateIi().new Solution();
    }
    
    // leetcode submit region begin(Prohibit modification and deletion)
    class Solution {
        public boolean containsNearbyDuplicate(int[] nums, int k) {
            Set<Integer> set = new HashSet<>();
            for (int i = 0; i < nums.length; i++) {
                if (!set.add(nums[i])) {
                    return true;
                }
                if (set.size() > k) {
                    set.remove(nums[i - k]);
                }
            }
            return false;
        }
    }
// leetcode submit region end(Prohibit modification and deletion)

}
       

File: leetcode/src/main/python/leetcode/editor/cn/存在重复元素IIcontainsDuplicateIi.py
"""
Title: 219.存在重复元素 II
Title Slug: containsDuplicateIi
Author: Neo
Date: 2025-01-29 10:51:16
"""
from typing import List


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        st = set()
        for i, x in enumerate(nums):
            if i > k:
                st.remove(nums[i - k - 1])
            if x in st:
                return True
            st.add(x)
        return False
# leetcode submit region end(Prohibit modification and deletion)

